docs/0010-differential-n-tuple-omit-prefix-storage-layout.md

```diff
@@ -0,0 +1,138 @@
+  * **OCFL Community Extensions Version:** 1.0
+  * **Obsoletes:** n/a
+  * **Obsoleted by:** n/a
+
+## Overview
+
+This OCFL storage root extension describes a storage layout combining a pairtree-inspired root directory structure intended to support identifiers with differential tuple sizes. For identifiers such as the [DRUID ("Digital Resource Unique IDentifier")](https://sdr.library.stanford.edu/documentation/purls-dois-and-orcid-ids), n-tuples in the identifier are of different sizes. For example, for the identifier `bc123df5678`, one would expect four tuples: `bc`, `123`, `df`, and `5678`.
+
+The OCFL object identifiers may contain prefixes which may be removed in the mapping to directory names. The OCFL object identifier prefix is defined as all characters before and including a configurable delimiter. The extension includes an option to include the full identifier as a leaf node.
+
+This layout is heavily based on the 0007-n-tuple-omit-prefix-storage-layout extension, adding the ability to handle tuples of different sizes.
+
+The limitations of this layout are filesystem dependent (with one exception), and are generally as follows:
+
+## Parameters
+
+### Summary
+
+* **Name:** `delimiter`
+  * **Description:** The case-insensitive delimiter marking the end of the OCFL object identifier prefix; MUST consist
+    of a character string of length one or greater. If the delimiter is found multiple times in the OCFL object
+    identifier, its last occurence (right-most) will be used to select the termination of the prefix.
+  * **Type:** string
+  * **Constraints:** Must not be empty
+  * **Default:** :
+* **Name**: `tupleSegments`
+  * **Description:** Indicates the tuple segment sizes (in characters) to split the identifier into
+  * **Type:** array
+  * **Constraints:** An array of integers, each of which corresponds to a tuple segment size
+  * **Default:** [2, 3, 2, 4]
+* **Name:** `fullIdentifierAsObjectRoot`
+  * **Description:** When true, indicates that the prefix-omitted, object identifier will be included as the leaf node of the layout, holding the object root.
+  * **Type:** boolean
+  * **Default:** false
+
+## Procedure
+
+The following is an outline of the steps to map an OCFL object identifier to an OCFL object root path:
+
+1. Remove the prefix, which is everything to the left of the right-most instance of the delimiter, as well as the delimiter. If there is no delimiter, the whole id is used; if the delimiter is found at the end, an error is thrown.
+1. Starting at the leftmost character of the resulting id and working right, divide the id into `tupleSegments`. If the length of the identifier does not equal the sum of the `tupleSegments`, an error is thrown.
+1. Create the start of the object root path by joining the tuples, in order, using the filesystem path separator.
+1. Optionally, if `fullIdentifierAsObjectRoot` is true, complete the object root path by joining the prefix-omitted id (from step 1) onto the end.
```

Comment:
I see this is missing from 007 but it might be clearer to say "onto the end \[after another filesystem path separator\]."